pkg/resource/training_job/sdk.go

```diff
@@ -1087,9 +1113,149 @@ func (rm *resourceManager) sdkUpdate(
 	desired *resource,
 	latest *resource,
 	delta *ackcompare.Delta,
-) (*resource, error) {
-	// TODO(jaypipes): Figure this out...
-	return nil, ackerr.NotImplemented
+) (updated *resource, err error) {
+	rlog := ackrtlog.FromContext(ctx)
+	exit := rlog.Trace("rm.sdkUpdate")
+	defer func() {
+		exit(err)
+	}()
+	input, err := rm.newUpdateRequestPayload(ctx, desired)
+	if err != nil {
+		return nil, err
+	}
+	warmpool_diff := delta.DifferentAt("Spec.ResourceConfig.KeepAlivePeriodInSeconds")
+	profiler_diff := delta.DifferentAt("Spec.ProfilerConfig") || delta.DifferentAt("Spec.ProfilerRuleConfigurations")
+	if warmpool_diff && profiler_diff {
+		return latest, ackerr.NewTerminalError(errors.New("cannot update Warm pool and Profiler at the same time"))
+	}
+	if !warmpool_diff && !profiler_diff {
+		return latest, ackerr.NewTerminalError(errors.New("only Warm Pool or Profiler can be updated"))
+	}
+	if warmpool_diff {
+		input.SetProfilerConfig(nil)
+		input.SetProfilerRuleConfigurations(nil)
+		warmpool_terminal := warmPoolTerminalCheck(latest)
+		if warmpool_terminal {
+			return latest, ackerr.NewTerminalError(errors.New("warm pool either does not exist or has reached a non updatable state"))
+		}
+		//Requeue if TrainingJob is in InProgress state
+		if err := customSetOutputUpdateWarmpool(latest); err != nil {
+			return nil, err
+		}
+	}
+	if profiler_diff {
+		if up_err := customSetOutputUpdateProfiler(latest); up_err != nil {
+			return nil, up_err
+		}
+		input.SetResourceConfig(nil)
+		if profilerRemovalCheck(desired, latest) {
+			handleProfilerRemoval(input)
+		} else {
+			inp_err := customSetUpdateInput(desired, latest, delta, input)
```

Comment:
can we move these methods handleProfilerRemoval, customSetUpdateInput etc. to hooks.go?

again, please make these methods part of resource manage class

Reply:
move both of those methods to hooks.go, the other methods in customsetinput.go modify the input so its better for them to stay there.